Tools/CISupport/ews-build/steps.py

```diff
@@ -653,9 +653,11 @@ class AnalyzeChange(buildstep.BuildStep):
 
     def _get_patch(self):
         sourcestamp = self.build.getSourceStamp(self.getProperty('codebase', ''))
-        if not sourcestamp or not sourcestamp.patch:
-            return None
-        return sourcestamp.patch[1]
+        if sourcestamp and sourcestamp.changes:
+            return '\n'.join(sourcestamp.changes[0].files)
+        if sourcestamp and sourcestamp.patch:
```

Comment:
Nit: let's keep the patch check before pr check.

Reply:
I did that originally. But I think that's actually the wrong approach. Suppose we had build that had both a patch and a list of files. We would want to use the list of files, no?

Reply:
We must not have both patch and PR. In that case, ValidateChange step would fail with error: "'Both patch_id and pr_number found. Unable to proceed with both" in https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L1159

>Suppose we had build that had both a patch and a list of files.
Did you notice any such build? Does patch based builds have sourcestamp.changes[0].files set?